### DISCUSSION TITLE
key-order for task become* before or after action?

### DISCUSSION THREAD

**Discussion by ssbarnea** [ansible/ansible-lint#2456] [Polls]
When having a task with keys like below, which order is better to use:

```
- name: ...
  become: ... # <-- should become* before action/module name?
  action: ...
  # or should become* be after the action?
  with_dict: ...
  notify: ...
```

That is important in relation to https://github.com/ansible/ansible-lint/pull/2454 which enables us to improve the sorting or task keys above requiring name to be the 1st key.

If you happen to also add an explanation why you would find one better than another, even better as we could improve the documentation.

**Comment by cidrblock** [comment]
I'm on the fence here, I prefer the name/plugin first as that is the most significant information, but "sudo" early is something I could get behind.

Slight pref for after...........

**Comment by apatard** [comment]
After thinking quite a lot about that, I don't care that much about the order but I've a small preference for putting it before the plugin. I often think about how to run a command (as root or as whatever): ``sudo <command>`` and so I tend to do the same for the task and put ``become*`` before the plugin. The other way may mean something like ``<command> sudo``.
